### ISSUE TITLE
Sound Blaster 16 installation program doesn't work

### ISSUE THREAD

**Issue by jschwartzenberg** [stsp/fdpp#113]
It doesn't work in FreeDOS either, but with MS-DOS 6.22 it runs fine. The errors suggest there is a difference in floppy drive handling between FreeDOS & MS-DOS that confuses the installer.

Download:
https://github.com/jschwartzenberg/dos-download-scripts/tree/master/sb16w31

**Comment by stsp** [comment]
Yes, quite easy to reproduce, but this
one does too many DOS calls. I check
the most suspects (like 44xx) but
nothing concrete.
Maybe Andrew can be more lucky. :)

**Comment by stsp** [comment]
I added dumping state on iret.
I think this is a great debugging capability
that will allow debugging "tough" things
like this one.

**Comment by andrewbird** [comment]
Something wrong with the last commit?

~~~
FAILED (failures=29, skipped=13)

The command "./travis_test.sh" exited with 1.

~~~

**Comment by stsp** [comment]
Yes, trying to find out why.

**Comment by stsp** [comment]
Should now be better.

**Comment by stsp** [comment]
Wow, with proper debugging it just took 10 minutes to fix!
Spent entire day yesterday on this. :(

**Comment by jschwartzenberg** [comment]
Indeed fixed! This allows installing Windows 3.1 and all drivers (video, mouse & audio) with FDPP.

**Comment by stsp** [comment]
Good to know, thanks for testing.
While freedos is incompatible and buggy,
the required efforts at fixing it are surprisingly
not very large. If dosemu1 code would be
that easy to maintain, we'd already be at
dosemu3 by now. :)
